skimage/feature/corner_cy.pyx

```diff
@@ -145,6 +145,19 @@ def _corner_fast(np_floats[:, ::1] image, signed char n, np_floats threshold):
                 lower_threshold = curr_pixel - threshold
                 upper_threshold = curr_pixel + threshold
 
+                # High speed test for n >= 12
+                if n >= 12:
+                    speed_sum_b = 0
+                    speed_sum_d = 0
+                    for k in range(0, 16, 4):
+                        ring_pixel = image[i + rp[k], j + cp[k]]
```

Comment:
Repeating the [question](https://github.com/scikit-image/scikit-image/pull/7379#discussion_r1557626440) here: 

@lagru:
> Considering the offset arrays `rp` and `cp` have 16 elements, how do these lines make sense. Shouldn't `rp[16]` be out of bounds and return a nonsensical result?

@spdfghi:
> [...] i am not native English speaker, so i might misunderstood the question, as range for in python is used like [a, b), range(0, 16, 4) is 0, 4, 8, 12

You are completely right! I totally missed, the relevance of the step parameter being 4.